.github/workflows/custom_actions.yaml

```diff
@@ -0,0 +1,67 @@
+name: custom_actions
+on:
+  push:
+    branches:
+      - main
+      - stable
+      - next
+    paths:
+      - "actions/**/*.dart"
+      - "actions/**/*.yaml"
+      - "actions/**/*.json"
+  pull_request:
+    paths:
+      - "actions/**/*.dart"
+      - "actions/**/*.yaml"
+      - "actions/**/*.json"
+  schedule:
+    - cron: "0 0 * * 0" # Every Sunday at 00:00
+defaults:
+  run:
+    shell: bash
+permissions: read-all
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # 3.5.3
+
+      - name: Setup Dart
+        uses: dart-lang/setup-dart@cdb51fff2b088939ef86fe041b18e82cb4733522 # main
```

Comment:
nit, not the same as other uses https://github.com/aws-amplify/amplify-flutter/blob/b01b29be58c47b8e9231122e7c0249081ec00541/packages/aft/workflow.yaml#L37